RecoMET/METFilters/plugins/HFNoisyHitsFilter.cc

```diff
@@ -0,0 +1,148 @@
+// -*- C++ -*-
+//
+// Package:    RecoMET/METFilters
+// Class:      HFNoisyHitsFilter
+//
+/**\class HFNoisyHitsFilter HFNoisyHitsFilter.cc RecoMET/METFilters/plugins/HFNoisyHitsFilter.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Laurent Thomas
+//         Created:  Tue, 01 Sep 2020 11:24:33 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+
+#include "DataFormats/HcalRecHit/interface/HFRecHit.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "DataFormats/METReco/interface/HcalPhase1FlagLabels.h"
+//
+// class declaration
+//
+
+class HFNoisyHitsFilter : public edm::global::EDFilter<> {
+public:
+  explicit HFNoisyHitsFilter(const edm::ParameterSet&);
+  ~HFNoisyHitsFilter() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  bool filter(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  std::vector<int> getNoiseBits() const;
+  const edm::EDGetTokenT<HFRecHitCollection> hfhits_token_;
+  const double rechitPtThreshold_;
+  const std::vector<std::string> listOfNoises_;
+  const bool taggingMode_;
+  const bool debug_;
+};
+
+//
+// constructors and destructor
+//
+HFNoisyHitsFilter::HFNoisyHitsFilter(const edm::ParameterSet& iConfig)
+    : hfhits_token_(consumes<HFRecHitCollection>(iConfig.getParameter<edm::InputTag>("hfrechits"))),
+      rechitPtThreshold_(iConfig.getParameter<double>("rechitPtThreshold")),
+      listOfNoises_(iConfig.getParameter<std::vector<std::string>>("listOfNoises")),
+      taggingMode_(iConfig.getParameter<bool>("taggingMode")),
+      debug_(iConfig.getParameter<bool>("debug")) {
+  produces<bool>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool HFNoisyHitsFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  using namespace edm;
+  bool pass = true;
+
+  // Calo Geometry - needed for computing E_t
+  ESHandle<CaloGeometry> pG;
+  iSetup.get<CaloGeometryRecord>().get(pG);
+  const CaloGeometry* geo = pG.product();
+
+  edm::Handle<HFRecHitCollection> theHFhits;
+  iEvent.getByToken(hfhits_token_, theHFhits);
```

Comment:
`the` is typically used for the data members, better not make it possibly confused with a local.
Also, the handle itself seems unnecessary, get the hit collection directly
```suggestion
  auto const& hfHits iEvent.get(hfhits_token_);
```